library/nmstate_interface.py

```diff
@@ -52,8 +52,37 @@
 class AnsibleNMStateInterface(AnsibleNMState):
     def run(self):
         name = self.params["name"]
-        interface_state = get_interface_state(self.interfaces, name)
-        interface_state["state"] = self.params["state"]
+        full_interface_state = get_interface_state(self.interfaces, name)
+
+        interface_state = {}
+        for key in ["type", "name", "mtu", "ethernet", "description"]:
+            if key in full_interface_state:
+                interface_state[key] = full_interface_state[key]
+
+        state = self.params["state"]
```

Comment:
How about hiding this implementation behind a small function/method? update_interface_state()
I would do that for all the others as well, cleaning this method from details.

Reply:
Should only need type and name here, therefore I simplified it.